tests/unit/button/button_options.js

```diff
@@ -6,152 +6,101 @@
+	deepEqual( element.prop( "disabled" ), true, "Disabled property is set" );
 });
 
-test("disabled, null", function() {
+// We are testing the default here because the default null is a special value which means to check
+// the DOM, so we need to make sure this happens correctly checking the options should never return
+// null. It should always be true or false
+test( "disabled, null", function() {
 	expect( 4 );
-	$("#radio01").button({ disabled: null });
-	deepEqual(false, $("#radio01").button("option", "disabled"),
+	var element = $( "#button" ),
+		elementDisabled = $( "#button-disabled" );
+	element.add( elementDisabled ).button({ disabled: null });
+	strictEqual( element.button("option", "disabled"), false,
 		"disabled option set to false");
-	deepEqual(false, $("#radio01").prop("disabled"), "element is disabled");
-
-	$("#radio02").prop("disabled", true).button({ disabled: null });
-	deepEqual(true, $("#radio02").button("option", "disabled"),
-		"disabled option set to true");
-	deepEqual(true, $("#radio02").prop("disabled"), "element is not disabled");
-});
-
-test( "disabled, ui-state-active is removed unless checkbox or radio", function( assert ) {
-	expect( 12 );
-	var elements = [
-		$( "<input type='button'>" ),
-		$( "<button></button>" ),
-		$( "<a></a>" ),
-		$( "<div></div>" ),
-		$( "<input type='checkbox' id='checkbox' checked><label for='checkbox'></label>" ),
-		$( "<input type='radio' id='radio' checked><label for='radio'></label>" )
-	];
-
-	$.each( elements, function() {
-		var element = $( this ).first().button(),
-			buttonElement = element.button( "widget" ),
-			elementType = element.prop( "nodeName" ).toLowerCase();
-
-		if ( element.is( "input" ) ) {
-			elementType += ":" + element.attr( "type" );
-		}
-
-		element.trigger( "mousedown" );
-		assert.hasClasses( buttonElement, "ui-state-active",
-			"[" + elementType + "] has ui-state-active class after mousedown." );
-
-		element.button( "disable" );
-		if ( element.is( "[type=checkbox], [type=radio]" ) ) {
-		assert.hasClasses( buttonElement, "ui-state-active",
-				"Disabled [" + elementType + "] has ui-state-active class." );
-		} else {
-			assert.lacksClasses( buttonElement, "ui-state-active",
-				"Disabled [" + elementType + "] does not have ui-state-active class." );
-		}
-	});
+	strictEqual( element.prop("disabled"), false, "element is disabled");
+	strictEqual( elementDisabled.button("option", "disabled"), true,
+		"disabled option set to false");
+	strictEqual( elementDisabled.prop("disabled"), true, "element is disabled");
 });
 
-test("text false without icon", function() {
+test( "showLabel, false, without icon", function( assert ) {
 	expect( 1 );
-	$("#button").button({
-		text: false
+
+	var button = $( "#button" );
+	button.button({
+		showLabel: false
 	});
-	ok( $("#button").is(".ui-button-text-only:not(.ui-button-icon-only)") );
+	assert.hasClasses( button, "ui-corner-all ui-widget" );
```

Comment:
This looks a bit naive to verify that there's nothing related to icons. `strictHasClasses` might be an option.